numba/parfors/parfor_lowering_utils.py

```diff
@@ -0,0 +1,211 @@
+from collections import namedtuple
+
+from numba.core import types, ir
+from numba.core.ir_utils import mk_unique_var
+from numba.core.typing import signature
+
+
+_CallableNode = namedtuple("BoundFunc", ["func", "sig"])
+
+
+class ParforLoweringBuilder:
```

Comment:
Docstring?

Reply:
Also wondering if this needs to be specifically for parfors, the utilities herein are generally useful if having to assemble IR manually? Perhaps because of the immediate lowering it prevents reuse at present.

Reply:
Added docstring.

I thought about making it more general. But, I'd rather do that incrementally.